### DISCUSSION TITLE
"slither ." command not working

### DISCUSSION THREAD

**Discussion by EthanHuntIMF99** [crytic/slither#2339] [Q&A] [answered]
please help me out this is the error that I am getting:

**slither .
'forge clean' running (wd: /home/stark99/foundry-stablecoin-f23)
'forge' returned non-zero exit code 1
FORGE - training program for SNAP (version 2006-07-28)
stderr: 
stderr: usage: forge [options] <ann> <dna> [options]
stderr: options:
stderr:   -help
stderr:   -verbose
stderr:   -pseudocount <float>  [1]   (absolute number for all models)
stderr:   -pseudoCoding <float> [0.0] (eg. 0.05)
stderr:   -pseudoIntron <float> [0.0]
stderr:   -pseudoInter <float>  [0.0]
stderr:   -min-counts           [0]
stderr:   -lcmask [-fragmentN]
stderr:   -utr5-length <int>    [50]
stderr:   -utr5-offset <int>    [10]
stderr:   -utr3-length <int>    [50]
stderr:   -utr3-offset <int>    [10]
stderr:   -explicit <int>       [250]
stderr:   -min-intron <int>     [30]
stderr:   -boost <file>  (file of ID <int>)
'forge config --json' running
ZOE ERROR (from forge): zoeParseOptions: unknown option (--json)
ZOE library version 2013-02-16
Traceback (most recent call last):
  File "/home/stark99/.local/bin/slither", line 8, in <module>
    sys.exit(main())
  File "/home/stark99/.local/lib/python3.10/site-packages/slither/__main__.py", line 727, in main
    main_impl(all_detector_classes=detectors, all_printer_classes=printers)
  File "/home/stark99/.local/lib/python3.10/site-packages/slither/__main__.py", line 833, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/home/stark99/.local/lib/python3.10/site-packages/slither/__main__.py", line 96, in process_all
    compilations = compile_all(target, **vars(args))
  File "/home/stark99/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 718, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
  File "/home/stark99/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 207, in __init__
    self._compile(**kwargs)
  File "/home/stark99/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 629, in _compile
    self._platform.compile(self, **kwargs)
  File "/home/stark99/.local/lib/python3.10/site-packages/crytic_compile/platform/foundry.py", line 66, in compile
    foundry_config = self.config(self._target)
  File "/home/stark99/.local/lib/python3.10/site-packages/crytic_compile/platform/foundry.py", line 133, in config
    subprocess.run(
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['forge', 'config', '--json']' returned non-zero exit status 2.**

![image](https://github.com/crytic/slither/assets/142003515/2d945e0c-6a42-4522-910b-cac0dae8a6c4)

PS: I am using Foundry for this project and I have tried every possible method including using the local environment variable etc. If anyone had the same problem and they have sorted it , please lemme know and help me out.

**Comment by 0xalpharush** [comment]
It looks like you have another program called `forge` that is not the one from Foundry. You can install the correct one here https://book.getfoundry.sh/getting-started/installation

**Reply by EthanHuntIMF99** [reply]
If that was the problem could it have compiled and tested the contracts(because rn I can compile and test the contracts perfectly). I mean is that possible? idk I have doubt regarding this cuz after installing foundry and stuff I have to run one command so that I can use foundry's functionalities and that is :alias forge='~/.foundry/bin/forge' . And I can perform every forge command that is mentioned in foundrybook. I believe that is not the problem I appreciate you reaching out to help but I'm afraid that is not the case I guess. Any other suggestions would be appreciated I have to present the project tomorrow I hope I can get the help ASAP thanx in adv ;)

**Reply by elopez** [reply, marked as answer]
executing programs do not involve the shell and therefore do not take into account aliases, but you can probably tweak your `PATH` so that Foundry's forge takes precedence over this other `forge` you have. Try running this before using Slither:

```sh
export PATH="$HOME/.foundry/bin:$PATH"
```

Alternatively, uninstall that other `forge` program you have. It appears to be included in the `snap` package from Ubuntu; maybe you installed that by mistake in the past? It seems to be some software to work with DNA sequences.

**Reply by EthanHuntIMF99** [reply]
this "PATH" thing worked man thnx a lot dude I owe you one
